### PR TITLE
Runtime log collection

### PR DESCRIPTION
**Pull request by kn-ms**
I have added an optional feature to collect logs (in separate ouput channel) of the test run. In testplan it is not possible to collect per testcase logs, but the execution logs may contains usefull information.

- The feature is configurable from settings
- one output channel is created per test adapters, and reused between runs
- optionally the output channel get visible on start of the test run, so the log stream can be seen.

Additionally I have refactored some of the code dupplication in unittests. Introduced sinon mocking lib, so some more test can be simplified with mocking later.

This pr is on top of #299 so best to view diffs when that get merged, and this get rebased.

### CODE REVIEW

**Comment by kn-ms** [comment]
@kondratyev-nv I have rebased on master now, it should be easier to check the diffs. (It is still quiet bit of a change)

File: src/pythonTestAdapter.ts [hunk 1]
```diff
@@ -136,9 +149,21 @@ export class PythonTestAdapter implements TestAdapter {
         try {
             this.testStatesEmitter.fire({ type: 'started', tests });
             const config = await this.configurationFactory.get(this.workspaceFolder);
+
+            let collector: LoggingOutputCollector | undefined;
```
Comment (kondratyev-nv) [inline src/pythonTestAdapter.ts:153]: nit: if you extract initialization to a function you can use const and code might be a bit cleaner.

**Review by kondratyev-nv — APPROVED** [review]
Can you please add some screenshots or examples how output looks like?

File: src/pythonTestAdapter.ts (hunk 1 above)
Comment (kn-ms) [reply src/pythonTestAdapter.ts]: I have extracted it to function

**Comment by kn-ms** [comment]
Here are some screenshots how it looks with teh different frameworks. @kondratyev-nv If you like it and it get merged, can you publish a version as well, I know at least one guy who waiting to have his hands on this. Thx K

testplan-1694982711298
image
image

**Pull request merged**